src/sql/workbench/contrib/notebook/test/browser/outputProcessor.test.ts

```diff
@@ -0,0 +1,136 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// import * as TypeMoq from 'typemoq';
+import * as assert from 'assert';
+import { nb } from 'azdata';
+import * as op from 'sql/workbench/contrib/notebook/browser/models/outputProcessor';
+import { JSONObject } from 'sql/workbench/services/notebook/common/jsonext';
+import { nbformat as nbformat } from 'sql/workbench/services/notebook/common/nbformat';
+import { getRandomString } from 'vs/editor/test/common/model/linesTextBuffer/textBufferAutoTestUtils';
+
+
+suite('OutputProcessor functions', function (): void {
+	suite('getData', async function (): Promise<void> {
+		for (const outputType of ['execute_result', 'display_data', 'update_display_data', 'stream', 'error'] as nbformat.OutputType[]) {
+			const text = getRandomString(1, 10);
+			const output: nb.ICellOutput = {
+				output_type: outputType
+			};
+			if (nbformat.isError(output)) {
+				for (const traceback of [undefined, [], [getRandomString(1, 10), getRandomString(1, 10)]]) {
+					for (const evalue of [undefined, getRandomString(1, 10)]) {
+						for (const ename of [text, '']) {
+							test(`test for outputType:${outputType}, ename:'${ename}', evalue:${evalue}, and traceback:${JSON.stringify(traceback)}`, async () => {
+								output.ename = ename;
+								output.evalue = evalue;
+								output.traceback = traceback;
+								const result = op.getData(output);
+								const tracedata = (traceback === undefined || traceback === []) ? undefined : traceback.join('\n');
+								let expectedData: JSONObject = {
+									'application/vnd.jupyter.stderr': undefined
+								};
+								if (tracedata) {
+									expectedData = {
+										'application/vnd.jupyter.stderr': tracedata
+									};
+								} else if (evalue) {
+									if (ename !== '') {
+										expectedData = {
+											'application/vnd.jupyter.stderr': `${output.ename}: ${output.evalue}`
+										};
+									} else {
+										expectedData = {
+											'application/vnd.jupyter.stderr': `${output.evalue}`
+										};
+									}
+
+								}
+								assert.deepEqual(result, <JSONObject>expectedData, `getData should return the expectedData:'${JSON.stringify(expectedData)}' object`);
+							});
+						}
+					}
+				}
+			} else if (nbformat.isStream(output)) {
+				for (const name of ['stderr', 'stdout'] as nbformat.StreamType[]) {
+					test(`test for outputType:${outputType} and name:${name}`, async () => {
+						output.text = text;
+						output.name = name;
+						const expectedData =
```

Comment:
Another example of being way overly verbose. If you're going to use loops why not just embed the expected values into the loop params?
```
for (const item of [ 
   'stderr', { 'application/vnd.jupyter.stderr': text},
   'stdout', { 'application/vnd.jupyter.stdout', text}
]) {
...
```
That seems much cleaner to me and clearly shows what the expected values are in the code for each type.

Reply:
This was a good suggestion but I ended up doing a different refactor where I felt it was not needed (but could still be done).